rte/mo_rte_lw.F90

```diff
@@ -178,6 +182,17 @@ function rte_lw(optical_props, top_at_1, &
       return
     end if
 
+    !
+    ! Check extents and values of transport angle secants if supplied
+    !
+    if (present(lw_Ds)) then
+      if(.not. extents_are(lw_Ds, ncol, ngpt)) &
+        error_msg = "rte_lw: lw_Ds inconsistently sized"
+      if(.not. any_vals_less_than(lw_Ds, 0._wp)) &
```

Comment:
I think you don't want this not: you're checking that all the Ds are > 0, no?

Reply:
I did not add this code...says "blame"...From Eli, "we are doing a linear function of T.  so the values for T=0 and T=1 have to be reasonable.  T=0 is the second coefficient and T=1 is their sum.  so one could check that the second and the sum are both ge > 1.  i can think of no circumstances in which they’d be > 2, so I suppose one could also check that both the first and the sum are < 2."

Reply:
Eli also says, "i’m assuming that these are just sanity checks for bad values, so my proposed tests, although soft, will find insane values.  if we wanted tighter tests, then it looks like having both the second and the sum between 1.5 and 2 would work."

Reply:
My suggestion is not to be too clever, but simply check that all `lw_Ds` are > 0.

Reply:
I think the point is some of the lw_Ds will be 0 or less than 0.  I'll add eli's logic.

Reply:
@jdelamere The lw_Ds are the secants of the propagation angle, one per g-point and column. Eli's logic refers to coefficient of linear fit that produce these secants and doesn't apply here. Or have I missed something?

Reply:
Minor question here regarding style - you have routines "any_vals_less_than" - if we need less than, or equal to, do we set up a new set of functions?

Reply:
If you're testing the secant it seems like "less than 1" works, but you could also check for any_vals_less_than( val + spacing(val)). That's probably preferable to developing a whole new set of routines.